src/ck3/data/lease_contracts.rs

```diff
@@ -0,0 +1,86 @@
+use crate::block::{Block, BV};
+use crate::context::ScopeContext;
+use crate::db::{Db, DbKind};
+use crate::everything::Everything;
+use crate::game::GameFlags;
+use crate::item::{Item, ItemLoader};
+use crate::scopes::Scopes;
+use crate::token::Token;
+use crate::tooltipped::Tooltipped;
+use crate::trigger::validate_trigger;
+use crate::validate::validate_modifiers_with_base;
+use crate::validator::{Validator, ValueValidator};
+
+#[derive(Clone, Debug)]
+pub struct LeaseContract {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Ck3, Item::LeaseContract, LeaseContract::add)
+}
+
+// TODO: verify somewhere that `theocracy_lease` is a defined item.
+
+impl LeaseContract {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::LeaseContract, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for LeaseContract {
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        let mut vd = Validator::new(block, data);
+        vd.req_field("hierarchy");
```

Comment:
> "Definition for automatic lease hierarchy. Required for some built-in lease contracts, not supported for the others."
best to only require and allow for the built-in ones

Reply:
Oh I see

I don't read the .infos carefully enough